### ISSUE TITLE
Unifi Core Backups Not Supported

### ISSUE THREAD

**Issue by andyad8865** [zhangyoufu/unifi-backup-decrypt#4]
Backups from unifi core ending in .unifi do not decrypt using this tool, this includes cloud backups.
all that is given is an empty zip file of around 22 bytes.

**Comment by zhangyoufu** [comment]
I don't own UDM Pro, nor do I have access to Cloud Console. That being said, I'm not able to add support for `.unifi` file format in short term.

I'll leave this issue open, any contribution is welcomed.

**Comment by rhcp011235** [comment]
Yep, Having the same issue here. I thought this would help. many people have linked it. @zhangyoufu if I shared a file with you? would you be willing to check it out?

**Comment by zhangyoufu** [comment]
@rhcp011235 

From my limited understanding, the `.unifi` backup files are encrypted using a key pair and SSO id, which means there is no single credential that could decrypt all `.unifi` backup archive. It differ from `.unf` files a lot.

You can checkout `/usr/share/unifi-core/app/node_modules/@ubnt/encrypted-archive/v1.js` and `/usr/share/unifi-core/app/service.js` for details.

I don't own UDM device, so I am not able to verify on my own. That's it.

**Comment by rhcp011235** [comment]
> [@rhcp011235](https://github.com/rhcp011235)
> 
> From my limited understanding, the `.unifi` backup files are encrypted using a key pair and SSO id, which means there is no single credential that could decrypt all `.unifi` backup archive. It differ from `.unf` files a lot.
> 
> You can checkout `/usr/share/unifi-core/app/node_modules/@ubnt/encrypted-archive/v1.js` and `/usr/share/unifi-core/app/service.js` for details.
> 
> I don't own UDM device, so I am not able to verify on my own. That's it.

I can send you my /usr/share/unifi-core/app/node_modules/@ubnt/encrypted-archive/v1.js (It looks like what you need)

Can you email me @ John.b.hale@gmail.com

the second file doesnt seem to be much, I can export it and send too if needed, but the crypto stuff seems to be in the first one

thank you!

**Comment by rhcp011235** [comment]
Here is v1.js

https://pastebin.com/i7bfuDPn

Looks like it hashes the UDID + password. but it also uses some public key + pepper. kinda hard to figure out

**Comment by wkearney99** [comment]
Any progress on this front?  I've got a Cloud Gateway Fiber that has a problematic user record left over from a site import and I'd like to see about hacking it out of the backup database.  The user shows up as having "Custom" role, but cannot be deactivated or deleted.  I decrypted a regular unf backup and the record does not seem to be in there.
